dataset_template.py

```diff
@@ -0,0 +1,59 @@
+class Template:
+    """ Template object encapsulates a dictionary of template options.
+
+        Given a dataset.template file, this object creates a dictionary of options keyed on the option name. This
+        will allow for accessing of various template parameters in constant time (as opposed to O(n) time) and includes
+        a simple, user friendly syntax.
+
+        Use as follows:
+
+            template = Template(file_name)          # create a template object
+            options = template.get_options()        # access the options dictionary
+            dataset = options['dataset']            # access a specific option
+
+    """
+
+    options = {} # The options dictionary
+
+    def __init__(self, template_file):
+        """ Initializes Template object with a template file.
+
+            The provided template file is parsed line by line, and each option is added to the options dictionary.
+
+            :param template_file: file, the template file to be accessed
+        """
+
+        # Adds the dataset name as parsed from the filename to the dictionary for easy lookup
+        self.options['dataset'] = template_file.split('/')[-1].split(".")[0]
+
+        # Open files for reading
+        with open(template_file) as template:
+
+            for line in template:
+
+                if "####################" not in line:
+
+                    # Splits each line on the ' = ' character string
```

Comment:
Seems like a self-explanatory comment.

Reply:
Rather be explicit than let people make assumptions.

Reply:
Seems like a `Add 4 to bias` instance. https://www.quora.com/What-are-the-best-practices-for-commenting-code/answer/Dave-Wade-Stein